### PR TITLE
ISPN-7961 Cross-site replication of functional commands

### PR DESCRIPTION
**Pull request by rvansa**
https://issues.jboss.org/browse/ISPN-7961

This is updated #5232 rebased on top of #5227. I've used `WriteEntryView.set(Object, Metadata)` instead of converting `Metadata` -> `MetaParam.Writeable[]` -> `Metadata`, and updated the description in `NonTxBackupFailureTest`.

### CODE REVIEW

**Comment by pruivo** [comment]
needs rebase @rvansa

**Comment by rvansa** [comment]
@pruivo rebased

File: core/src/main/java/org/infinispan/functional/impl/MetaParamsInternalMetadata.java [hunk 1]
```diff
@@ -107,50 +107,58 @@ public String toString() {
          '}';
    }
 
+   public MetaParam.Writable[] toWritableMetas() {
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/functional/impl/MetaParamsInternalMetadata.java]: no longer used. can be removed

File: core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteConsumerTopologyChangeTest.java [hunk 2]
```diff
@@ -56,6 +56,7 @@ public void testXSiteDuringJoin() throws InterruptedException, ExecutionExceptio
       doXSiteStateTransferDuringTopologyChange(TopologyEvent.JOIN);
    }
 
+   @Test(groups = "unstable_xsite", description = "ISPN-6872")
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteConsumerTopologyChangeTest.java]: the class is disabled. no need to disable the test :)

**Comment by rvansa** [comment]
@pruivo addressed

**Comment by pruivo** [comment]
integrated! thanks @rvansa

**Pull request merged**